packages/grid/src/GridUtils.ts

```diff
@@ -739,6 +739,46 @@ export class GridUtils {
     return movedItems;
   }
 
+  /**
+   * Move a visible range in the grid
+   *
+   * This will effectively slice the range out of the grid,
+   * re-index the remaining columns,
+   * then insert the range with the first element at the provided index
+   *
+   * @param from The visible axis range to move
+   * @param to The visible index to move the start of the range to
+   * @param oldMovedItems The old reordered items
+   * @returns The new reordered items
+   */
+  static moveRange(
+    from: AxisRange,
+    to: VisibleIndex,
+    oldMovedItems: MoveOperation[] = []
+  ): MoveOperation[] {
+    if (from[0] === to) {
+      return oldMovedItems;
+    }
+
+    const movedItems: MoveOperation[] = [...oldMovedItems];
+    const lastMovedItem = movedItems[movedItems.length - 1];
+    if (
+      lastMovedItem &&
+      Array.isArray(lastMovedItem.from) &&
```

Comment:
Might be worth having an `isAxisRange` predicate (not just an array but an array of length 2)